edb/server/protocol/binary.pyx

```diff
@@ -1148,7 +1141,15 @@ cdef class EdgeConnection(frontend.FrontendConnection):
 
         state_tid = self.buffer.read_bytes(16)
         state_data = self.buffer.read_len_prefixed_bytes()
-        self.get_dbview().decode_state(state_tid, state_data)
+        if state_tid == sertypes.INVALID_TYPE_ID.bytes:
```

Comment:
I don't think we need this shortcut.  Most clients will receive correct state id on connect.